File: backend/app/schemas/statistics.py
from __future__ import annotations

from pydantic import BaseModel


class OverviewStats(BaseModel):
    total_lifts: int
    total_lift_types: int
    total_customers: int
    total_contracts: int
    active_contracts: int
    total_revenue: float
    total_cost: float
    net_profit: float


class MonthlyStat(BaseModel):
    year: int
    month: int
    label: str
    revenue: float
    cost: float
    profit: float
    contracts: int


class YearlyStat(BaseModel):
    year: int
    revenue: float
    cost: float
    profit: float
    contracts: int


class LiftStat(BaseModel):
    lift_id: int
    uid: str
    name: str
    lift_type_name: str | None = None
    revenue: float
    cost: float
    profit: float
    contracts: int


class LiftStatDetail(LiftStat):
    monthly: list[MonthlyStat] = []
    yearly: list[YearlyStat] = []

File: frontend/src/types/statistics.ts
export interface OverviewStats {
  total_lifts: number
  total_lift_types: number
  total_customers: number
  total_contracts: number
  active_contracts: number
  total_revenue: number
  total_cost: number
  net_profit: number
}

export interface MonthlyStat {
  year: number
  month: number
  label: string
  revenue: number
  cost: number
  profit: number
  contracts: number
}

export interface YearlyStat {
  year: number
  revenue: number
  cost: number
  profit: number
  contracts: number
}

export interface LiftStat {
  lift_id: number
  uid: string
  name: string
  lift_type_name: string | null
  revenue: number
  cost: number
  profit: number
  contracts: number
}

export interface LiftStatDetail extends LiftStat {
  monthly: MonthlyStat[]
  yearly: YearlyStat[]
}
